src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,34 +90,29 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
+        // If the configuration has not changed, return the current instance without validation
+        // since we've already validated and/or errored about the data
+
+        DatadogHttpClient newInstance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        DatadogHttpClient httpInstance = (DatadogHttpClient) instance;
+        if (httpInstance != null && httpInstance.equals(newInstance)) {
+            if (DatadogHttpClient.failedLastValidation) {
```

Comment:
We will need a way to retry validation every X minutes. What if `failedLastValidation` is true because of a network failure ? You'd require a restart of Jenkins to fix

Reply:
Hmm that's true. I will look into a way to do this.

Reply:
We still need this to be addressed

Reply:
I know, I haven't thought of a great approach yet. Do you have any ideas?

Reply:
So, in the case of a network failure, it wouldn't require a restart of jenkins, it would just require entering different credentials (and then entering the correct ones). It isn't the best solution, but I'm not sure how likely this scenario is.